### PR TITLE
NCT - Commitments (major draft)

### PR DESCRIPTION
**Pull request by Zorlin**
* Adds all DST commitments with detailed descriptions and maps to Narratives
* Adds some basic early tasks

Ready for review.

### CODE REVIEW

File: content/dst/codex/codex-scaling.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+`vac:dst:codex:codex-scaling`
+
+## Description
+Use real world testing, theoretical analysis and simulation to determine and improve Codex's scaling properties. Find the limits of Codex's capabilities and measure them in different scenarios.
+
+We want to learn as much as we can about Codex's behaviour and how that maps to our theory work. We want to learn what new theory work we can do from what we learn in the process.
+
+We will use a combination of real world testing, theoretical analysis and simulation to determine and improve Codex's scaling properties.
```
Comment (AlbertoSoutullo) [inline content/dst/codex/codex-scaling.md:18]: This is repeated from the first sentence

File: content/dst/nomos/nomos-scaling.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+`vac:dst:nomos:nomos-scaling`
+
+## Description
+Use real world testing, theoretical analysis and simulation to determine and improve Nomos's scaling properties. Find the limits of Nomos's capabilities and measure them in different scenarios.
+
+We want to learn as much as we can about Nomos's behaviour and how that maps to our theory work. We want to learn what new theory work we can do from what we learn in the process.
+
+We will use a combination of real world testing, theoretical analysis and simulation to determine and improve Nomos's scaling properties.
```
Comment (AlbertoSoutullo) [inline content/dst/nomos/nomos-scaling.md:18]: Same as the other comment

File: content/dst/waku/waku-scaling.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+---
+
+`vac:dst:waku:waku-scaling`
+
+## Description
+Use real world testing, theoretical analysis and simulation to determine and improve Waku's scaling properties. Find the limits of Waku's capabilities and measure them in different scenarios.
+
+We want to learn as much as we can about Waku's behaviour and how that maps to our theory work. We want to learn what new theory work we can do from what we learn in the process.
+
+We will use a combination of real world testing, theoretical analysis and simulation to determine and improve Waku's scaling properties.
```
Comment (AlbertoSoutullo) [inline content/dst/waku/waku-scaling.md:18]: Same

**Review by AlbertoSoutullo — APPROVED** [review]
Some nitpick comments

File: content/dst/nomos/nomos-scaling.md (hunk 2 above)
Comment (Zorlin) [reply content/dst/nomos/nomos-scaling.md]: Fixed

File: content/dst/waku/waku-scaling.md (hunk 3 above)
Comment (Zorlin) [reply content/dst/waku/waku-scaling.md:18]: Fixed

File: content/dst/waku/waku-scaling.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+- [ ] The monitoring system stays stable the entire time, providing useful information and metrics.
+
+### Test Store Protocol At Scale
+
+* fully qualified name: `vac:dst:waku:waku-scaling:test-store-protocol-at-scale`
+* owner: Alberto
+* status: 0%
+* start-date: 2024/09/02
+* end-date: 2024/12/31
```
Comment (AlbertoSoutullo) [inline content/dst/waku/waku-scaling.md:61]: If I am not wrong, all milestones that fall under the Waku scaling milestone are due to 2024/11/1. 
I mean Bandwidth, Reliability, Discv5, Store, Shards, Gowaku+Nwaku, Filter+Lightpush, peer exchange protocol and resource-restricted devices.

Comment (Zorlin) [reply content/dst/waku/waku-scaling.md:61]: We should push to make a schedule/timeline that fits properly.

File: content/dst/codex/codex-comparison.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  - "dst"
+  - "codex"
+draft: false
+description: "Measure Codex against systems like IPFS, BitTorrent etc and see how it compares. Primarily BitTorrent."
+---
+
+`vac:dst:codex:codex-comparison`
+
+## Description
+We will compare Codex to other systems like IPFS and BitTorrent to see how it performs. We will compare on things such as:
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/codex/codex-comparison.md]: Resolved, thanks, ready for re-review

File: content/dst/codex/codex-comparison.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+* Time to first byte
+* Bandwidth usage
+* Stability
+* Reliability
+
+Most importantly we will do a head to head speed test comparing download speeds of Codex against other systems. This will allow us to understand where Codex needs improvement and where it stands.
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/codex/codex-comparison.md]: Resolved, thanks, ready for re-review

File: content/dst/codex/codex-comparison.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+We will support the Conduit of Expertise narrative directly by providing valuable insights to Codex that allow them to understand how Codex performs in comparison to common and popular systems in the "altruistic" space.
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/codex/codex-comparison.md]: Resolved, thanks, ready for re-review

File: content/dst/codex/codex-comparison.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+These efforts will contribute in these ways to the Conduit of Expertise narrative:
+
+* Accelerate Codex reaching competitiveness with BitTorrent or find out what is and isn't possible to do.
+* Test the reliability of Codex in automated and highly stressful benchmarks that push its limits and reveal them in the process.
+* Improve the RFC culture by allowing us to reuse the work we do here to build future scenarios that can test complicated situations and requirements in a repeatable way.
+
+## Task List
+
+### > Matrices Deployments
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/codex/codex-comparison.md]: Fixed, thanks

File: content/dst/ift/visualiser-tool.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
+---
+title: Visualiser Tool
+tags:
+  - "2024q4"
+  - "dst"
+  - "ift"
+draft: false
+description: "Develop tools or frameworks suitable for visualising the state of arbitrary distributed systems and simulations. This initial iteration must support Waku visualisation, but future intention is to support any system which is log compatible with the Visualiser Tools."
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/ift/visualiser-tool.md]: We should probably not be mentioning the future intention in this one. I rewrote this, and will consider a further rewrite later. Please review the updated version :pray:

File: content/dst/ift/visualiser-tool.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
+---
+title: Visualiser Tool
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/ift/visualiser-tool.md:2]: We have prior existing work - in a half completed state - all of it is part of Q4 and the commitment. Not an issue :)

But the description should indeed make it clearer what we have. Noted and I will change that now. Cheers.

File: content/dst/vac/libp2p-evaluation.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+draft: false
+description: "Test libp2p on a regular basis and look for regressions, learn scaling properties and run scaling studies, understand the limits of libp2p and its behaviour."
+---
+
+`vac:dst:vac:libp2p-evaluation`
+
+## Description
+Test libp2p on a regular basis and look for regressions, learn scaling properties and run scaling studies, understand the limits of libp2p and its behaviour.
+
+We want to learn as much as we can about libp2p's behaviour and how it is evolving over time with each new release and with each thing we are specifically asked to check and test.
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/vac/libp2p-evaluation.md]: I've changed that to "Deliver hard numbers and actionable insights."

We should work out ASAP what those should be, but this at least captures an actionable item.

File: content/dst/waku/waku-scaling.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
+---
+title: Waku Scaling
+tags:
+  - "2024q4"
+  - "dst"
+  - "waku"
+draft: false
+description: "Use real world testing, theoretical analysis and simulation to determine and improve Waku's scaling properties. Find the limits of Waku's capabilities and measure them in different scenarios."
```
[the remark this answers is not in the document]
Comment (Zorlin) [reply content/dst/waku/waku-scaling.md:8]: We will finalize all work that is scheduled,

with the intention being that we will make this a recurring thing. Waku will continue to need scaling studies and analysis.

But we should definitely make sure everything captured in this Q4 cycle is realistically achievable with a defined end state.

**Comment by Zorlin** [comment]
@kaiserd @AlbertoSoutullo All comments addressed, ready for new review. Second pass complete, apologies for the delays.

I am going to continue working on this for a little while tonight as my intent is actually to turn this into an entire complete plan from start to finish, although for now I am focusing explicitly on only the parts of that required to complete this NCT update to our roadmap, per @kaiserd's wishes (and frankly, what makes sense :+1: )

**Comment by Zorlin** [comment]
Forgot a commit, added it now.

I'm going to start fleshing out the entire NCT structure now in Plane as all I have left to do before you guys give feedback is set timelines and specific dates, and I think we'll be able to use Plane to get a better idea of those.

**Pull request merged**